reactioncommerce/reaction
# chore: fix all `react/no-deprecated` and `react/prop-type` eslint warnings

Resolves #5247 & #5271    
Impact: **minor**  
Type: **chore**

## Issue
We currently have many `react/no-deprecated` and `react/prop-type` eslint warnings.

## Solution
Fix (or disable) these warnings, and then update our `eslint` rules to make `react/no-deprecated` and `react/prop-type` an `error` so new violations are not introduced.

I have update the components to use the `UNSAFE_` version fo deprecated react life cycle methods, as these components will be updated to be functional components in the future.

Updated rules: https://github.com/reactioncommerce/reaction-eslint-config/pull/15

## Breaking changes
None.

## Testing
1. run eslint
1. see there are no longer any `react/no-deprecated` and `react/prop-type` warnings (and no errors)
